### ISSUE TITLE
weird selection behavior in configuration editor

### ISSUE THREAD

**Issue by seebs** [sourcegraph/issues#57]
- Issue type: bug report
- Sourcegraph version: 2.6.8
- OS Version: ubuntu 16.04
- Docker version:
```Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.6.2
 Git commit:   092cba3
 Built:        Thu Nov  2 20:40:23 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.1
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.6.2
 Git commit:   092cba3
 Built:        Thu Nov  2 20:40:23 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

Steps to reproduce:

1. Select text in selection editor.
2. Middle-click in another window, expecting selected text.

This is specific to the X11 behavior where selecting text implicitly copies into a hidden secondary clipboard that isn't the same clipboard used by ^X/^C/^V. The standard clipboard works, and the weird X clipboard works for text outside that editor, but the X thing doesn't work with that editor. (It just doesn't count as "selected" text, so the selection-buffer still holds the last thing selected elsewhere, even just the text outside that editor in that window.)

**Comment by sqs** [comment]
I am unable to repro the specific issue you mentioned (selecting text in the config editor DOES add it to my X clipboard). However, middle-clicking in the config editor does NOT insert the contents of my X clipboard, as I would expect. Just to confirm, are you experiencing both?

Does the X clipboard work for you on text in the editor widgets at https://microsoft.github.io/monaco-editor/playground.html?

**Comment by seebs** [comment]
I get the same behavior with the Monaco editor directly, which is no impact on the selection-buffer clipboard. (I'm testing by selecting text elsewhere, confirming that I can middle-click-paste it, then selecting text in this window, and going elsewhere and middle-clicking. I still get the previous text.)

Hmm. In general, do you have different behavior between the select-buffer and the ^C/^V buffer? If I ^C some text in one window, and then select text in another buffer, middle-click pastes the selected text and ^V pastes the ^C text. (If the selection counts. So for instance, if I do that in this window, it works.) I think some X environments have helper apps which sync those buffers.

**Comment by sqs** [comment]
>  do you have different behavior between the select-buffer and the ^C/^V buffer?

Yeah, they are two different clipboards/buffers. I'm not running something that syncs the clipboards.

I did notice that in Chrome, selecting text in the config editor/Monaco does add it to the X clipboard, but in Firefox (57.0.3), selecting text does not. Are you using Firefox?

As for how we could fix this in Sourcegraph, we could add an option to use a normal textarea instead of the Monaco editor for site config. Would you prefer that?

**Comment by sqs** [comment]
Got it. Thanks again for the report. We will close this when either we have a workaround or when an upstream issue has been acked.
